### PR TITLE
Pretrained gaugan

### PR DESCRIPTION
**Pull request by melisandeteng**
Results here https://github.com/cc-ai/omnigan/issues/169

### CODE REVIEW

File: omnigan/eval_metrics.py [hunk 1]
```diff
@@ -43,8 +43,8 @@ def iou(pred_im, gt_im):
     IoU for binary masks (0-1 values)
 
     Args:
-        pred_im ([type]): [description]
-        gt_im ([type]): [description]
+        pred_im ([type]): prediction
```
Comment (vict0rsch) [inline omnigan/eval_metrics.py:46]: should these be numpy arrays?

File: omnigan/generator.py [hunk 2]
```diff
@@ -60,9 +61,55 @@ def get_gen(opts, latent_shape=None, verbose=0, no_init=False):
...
+        weights = torch.load(opts.gen.p.path_G_pretrained)
+        new_params = G.painter.state_dict().copy()
+        for elem in new_params:
+            if match_up_spade_v2(elem) in weights.keys() and not elem.startswith(
+                "conv_img"
+            ):
+                if new_params[elem].size() == weights[match_up_spade_v2(elem)].size():
+                    new_params[elem] = weights[match_up_spade_v2(elem)]
+            # If we only want to match the upsampling layers (that have the  right dimentsion)
+            #  if elem.startswith("up"):
```
Comment (vict0rsch) [inline omnigan/generator.py:75]: should these comments be removed?

File: omnigan/generator.py [hunk 3]
```diff
@@ -60,9 +61,55 @@ def get_gen(opts, latent_shape=None, verbose=0, no_init=False):
...
+    if "module" in s_:
+        s_.remove("module")
+    if int(s_[1]) in range(3):
+        return (".").join(["up_" + str(int(s_[1]) + 1), *s_[2:-1], match_name(s_[-1])])
+    else:
+        return ""
+
+
+def match_up_spade_v2(s):
+    s_ = s.split(".")
```
Comment (vict0rsch) [inline omnigan/generator.py:102]: can you add a docstring with an example of why this is needed?

Comment (vict0rsch) [reply omnigan/generator.py:102]: is the "v2" for spade (if so, what does it mean) or for the `match_up_spade` function? (if so, do you need the other version?)

**Review by vict0rsch — CHANGES_REQUESTED** [review]

File: omnigan/generator.py (hunk 3 above)
Comment (melisandeteng) [reply omnigan/generator.py:102]: Yes, i was trying out different ways of matching weights but I think I'll only keep one of them

**Comment by vict0rsch** [comment]
@melisandeteng should we merge this PR? if so can you update it? Thanks!

**Pull request closed without merging**